hudi-flink/src/main/java/org/apache/hudi/table/format/mor/MergeOnReadInputFormat.java

```diff
@@ -329,10 +330,9 @@ public boolean hasNext() {
 
               this.currentRecord =  delete;
               return true;
-            } else {
-              // delete record found, skipping
-              return hasNext();
             }
+            // skipping if the condition is unsatisfied
+            // continue;
```

Comment:
Can we delete this directly?

Reply:
we already did that, add this as a comment to make code more clear.

Reply:
OK~